### PR TITLE
Add GET /api/job/{jobid}/files/{path} route

### LINKED ISSUE #44: Add route to retrieve job output files
As someone who submitted a job I would like to get the output files of the job.

It would be nice if a HTML page in the job directory could be directly opened in web browser.
It should only show files of a job that the user owns.

### PR DESCRIPTION
**Pull request by sverhoeven**
Fixes #44

Also adds `GET /api/job/{jobid}/stderr` route.

To test
1. Execute job as described at https://github.com/i-VRESSE/bartender/tree/44-host-job-files#word-count-example
2. Try out the `GET /api/job/{jobid}/files/{path}` route
    * with path=returncode should return same content as `/tmp/jobs/1/returncode` file on filesystem.
3. Try out the `GET /api/job/{jobid}/stdout` route
    * should return same content as `/tmp/jobs/1/stdout.txt` file on filesystem.
4. Try out the `GET /api/job/{jobid}/stderr` route
    * should return same content as `/tmp/jobs/1/stderr.txt` file on filesystem.
5. Test nested paths
    1. on filesystem run `mkdir /tmp/jobs/1/output && echo foo > /tmp/jobs/1/output/somefile.txt`  
    2. Try `GET /api/job/{jobid}/files/output/somefile.txt`
    3. Should return `foo` as content

### CODE REVIEW

File: src/bartender/web/api/job/views.py [hunk 1]
```diff
@@ -87,45 +87,111 @@ async def retrieve_job(
...
+        HTTPException: When file is not found or is outside job directory.
+
+    Returns:
+        The file content.
+    """
+    try:
+        full_path = (job_dir / path).expanduser().resolve(strict=True)
+        if not full_path.is_relative_to(job_dir):
+            raise FileNotFoundError()
+        # TODO if path == '' or path is directory then return file listing?
```
Comment (Peter9192) [inline src/bartender/web/api/job/views.py:152]: #55

File: tests/conftest.py [hunk 2]
```diff
@@ -210,10 +209,38 @@ async def current_user_token(fastapi_app: FastAPI, client: AsyncClient) -> str:
...
 
     Returns:
         Headers needed for auth.
     """
     return {"Authorization": f"Bearer {current_user_token}"}
+
+
+@pytest.fixture
+async def second_user_token(fastapi_app: FastAPI, client: AsyncClient) -> str:
+    """Registers scond dummy user and returns its auth token.
```
Comment (Peter9192) [inline tests/conftest.py:237]:
```suggestion
    """Registers second dummy user and returns its auth token.
```

**Review by Peter9192 — APPROVED** [review]
Tested and this works well. See #55 for browsing directories

**Comment by sverhoeven** [comment]
Thanks for reviewing

**Pull request merged**